src/collects/seashell/backend/project.rkt

```diff
@@ -773,3 +775,93 @@
                                                               (read-config 'common-subdirectory)
                                                               question)
                                                           file)))))
+
+
+
+;; copy old dir/file to a new dir/file, handles parent creation correctly
+;; after copy children of old dir becomes children of new dir
+;; better than racket's file libaray function also gives an option to overwrite existing
+(define/contract (copy* old new overwrite)
+  (path-string? path-string? boolean? . -> . void?)
+  (when (equal? old new)
+    (raise-user-error "copy* cannot copy" old "to the same location" new))
+  (make-parent-directory* new)
+  (cond [(or (link-exists? old) 
+             (file-exists? old))
+          (when (or overwrite (not (file-exists? new)))
+            (copy-file old new))]
+        [(directory-exists? old) 
+          (for-each 
+            (lambda (item) 
+              (copy* (build-path old item) (build-path new item) overwrite))
+            (directory-list old))]
+        [else (raise-user-error (format "Could not copy ~a" old))]))
+
+
+;; unzip to a dest-dir with an option to overwrite existing
+(define/contract (unzip* zip-file dest-dir overwrite)
+  (path-string? path-string? boolean? . -> . void?)
+  (define zip-port (open-input-file zip-file))
+  (call-with-unzip zip-port (lambda (tmp-dir) (copy* tmp-dir dest-dir overwrite))))
```

Comment:
I would rather you implement the zip file reader contract and unzip instead of using call-with-unzip; you can look at restore-file-from-template to see how it works.